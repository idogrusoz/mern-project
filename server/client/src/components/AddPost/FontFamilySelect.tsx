import { MenuItem, Select } from "@material-ui/core";
import React, { Dispatch, FunctionComponent, SetStateAction } from "react";

type FontFamilySelectProps = {
    setFontFamily: Dispatch<SetStateAction<string>>;
    setFontWeightValues: Dispatch<SetStateAction<Array<number>>>;
};

interface fontInfo {
    label: string;
    value: string;
    options: number[];
}

const FontFamilySelect: FunctionComponent<FontFamilySelectProps> = ({ setFontFamily, setFontWeightValues }) => {
    const fontFamilies: fontInfo[] = [
        {
            label: "Caveat",
            value: "Caveat",
            options: [400, 700],
        },
        {
            label: "Dancing Script",
            value: "Dancing Script",
            options: [400, 700],
        },
        {
            label: "Kalam",
            value: "Kalam",
            options: [300, 400, 700],
        },
        {
            label: "Lemonada",
            value: "Lemonada",
            options: [300, 400, 700],
        },
        {
            label: "Lobster",
            value: "Lobster",
            options: [400, 700],
        },
        {
            label: "Montserrat",
            value: "Montserrat",
            options: [100, 400, 700, 900],
        },
        {
            label: "Montserrat Alternates",
            value: "Montserrat Alternates",
            options: [100, 400, 700, 900],
        },
        {
            label: "Permanent Marker",
            value: "Permanent Marker",
            options: [400],
        },
        {
            label: "Roboto",
            value: "Roboto",
            options: [100, 400, 900],
        },
        {
            label: "Sansita Swashed",
            value: "Sansita Swashed",
            options: [400, 700, 900],
        },
    ];

    const handleChange = (index: number) => {
        setFontFamily(fontFamilies[index].value);
        setFontWeightValues(fontFamilies[index].options);
    };

    return (
        <Select fullWidth defaultValue={8} onChange={(e) => handleChange(Number(e.target.value))}>
            {fontFamilies.map((font: fontInfo, i: number) => {
                return (
                    <MenuItem value={i} key={i}>
                        {font.label}
                    </MenuItem>
                );
            })}
        </Select>
    );
};

export default FontFamilySelect;
