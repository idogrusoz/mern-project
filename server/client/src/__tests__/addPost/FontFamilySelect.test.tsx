import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import FontFamilySelect from "../../components/addPost/FontFamilySelect";

describe("FontFamilySelect test", () => {
    const props = {
        fontFamily: "Roboto",
        setFontFamily: jest.fn(() => {}),
        setFontWeightValues: jest.fn(() => {}),
    };

    const wrapper = shallow(<FontFamilySelect {...props} />);
    it("renders without crash", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
