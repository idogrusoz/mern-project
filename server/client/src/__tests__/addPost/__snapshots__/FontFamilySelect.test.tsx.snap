// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontFamilySelect test renders without crash 1`] = `
<WithStyles(ForwardRef(Select))
  defaultValue={8}
  fullWidth={true}
  onChange={[Function]}
>
  <WithStyles(ForwardRef(MenuItem))
    key="0"
    value={0}
  >
    Caveat
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="1"
    value={1}
  >
    Dancing Script
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="2"
    value={2}
  >
    Kalam
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="3"
    value={3}
  >
    Lemonada
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="4"
    value={4}
  >
    Lobster
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="5"
    value={5}
  >
    Montserrat
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="6"
    value={6}
  >
    Montserrat Alternates
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="7"
    value={7}
  >
    Permanent Marker
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="8"
    value={8}
  >
    Roboto
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="9"
    value={9}
  >
    Sansita Swashed
  </WithStyles(ForwardRef(MenuItem))>
</WithStyles(ForwardRef(Select))>
`;
